AirportTicketBookingSystem.Tests/DomainTests/RepositoriesTests/BookingRepositoryTests.cs

```diff
@@ -0,0 +1,203 @@
+using Moq;
+using AutoFixture;
+using FluentAssertions;
+using Shouldly;
+using AirportTicketBookingSystem.Domain.Models;
+using AirportTicketBookingSystem.Domain.Repositories;
+using AirportTicketBookingSystem.Domain.Services;
+using AirportTicketBookingSystem.Domain.Utilities;
+
+public class BookingRepositoryTests
+{
+    private readonly Fixture _fixture;
+    private readonly IFileHandler _fileStorage;
+    private readonly Mock<IFlightService> _flightServiceMock;
+    private readonly BookingRepository _bookingRepository;
+
+    public BookingRepositoryTests()
+    {
+        _fixture = new Fixture();
+        _fileStorage = new FakeFileHandler();
+        _flightServiceMock = new Mock<IFlightService>();
+
+        _bookingRepository = new BookingRepository(_fileStorage, _flightServiceMock.Object);
+    }
+
+    [Fact]
+    public void AddBooking_Should_AddBookingToList()
+    {
+        var booking = _fixture.Create<Booking>();
+        //var booking = new Booking { Id = 1, Flight = new Flight { Id = 101 }, Passenger = new Passenger
+        //{ Id = 1, Name = "Mahmoud Miqdad" }, SeatClass = SeatClass.Economy, BookDate = DateTime.Now };
+
+        _bookingRepository.AddBooking(booking);
+
+        _bookingRepository.GetAllBookings().ShouldContain(booking);
+    }
+
+    [Fact]
+    public void AddBooking_Should_ThrowException_When_BookingAlreadyExists()
+    {
+        var booking = _fixture.Create<Booking>();
+
+        _bookingRepository.AddBooking(booking);
+        Action act = () => _bookingRepository.AddBooking(booking);
+
+        act.Should().Throw<InvalidOperationException>()
+            .WithMessage($"\nBooking {booking.Id} already exists");
+    }
+
+    [Fact]
+    public void GetBookingsByPassenger_Should_Return_CorrectBookings()
+    {
+        var booking = _fixture.Create<Booking>();
+
+        _bookingRepository.AddBooking(booking);
+        var result = _bookingRepository.GetBookingsByPassenger(booking.Passenger.Id);
+
+        result.Should().ContainSingle().Which.Should().Be(booking);
+    }
+
+    [Fact]
+    public void GetBookingsByPassenger_Should_ReturnEmptyList_When_NoBookingsForGivenPassenger()
```

Comment:
Lets stick to this naming convention `MethodName_StateUnderTest_ExpectedBehavior`. Same comment for the other tests